apps/meteor/packages/rocketchat-livechat/assets/rocket-livechat.js

```diff
@@ -672,16 +672,29 @@
 		title: null
 	};
 	var trackNavigation = function() {
-		setInterval(function() {
-			if (document.location.href !== currentPage.href) {
+		if (typeof navigation !== 'undefined'){
+			navigation.addEventListener('navigationSuccess', function(){
 				pageVisited('url');
 				currentPage.href = document.location.href;
-			}
-			if (document.title !== currentPage.title) {
-				pageVisited('title');
-				currentPage.title = document.title;
-			}
-		}, 800);
+			});
+			new MutationObserver(function(mutations) {
+				if (document.title !== currentPage.title){
+					pageVisited('title');
+					currentPage.title = document.title;
+				}
+			}).observe(document.querySelector('title'), { subtree: true, characterData: true, childList: true });
+		} else {
```

Comment:
Replace the document.querySelector('title') in the MutationObserver with a more robust method to ensure compatibility with edge cases where the <title> element might not be in the expected location. Instead, observe the head element for changes to the <title> to ensure broader compatibility:

`
new MutationObserver(function(mutations) {
    if (document.title !== currentPage.title) {
        pageVisited('title');
        currentPage.title = document.title;
    }
}).observe(document.querySelector('head'), { subtree: true, childList: true });

`

Reason:
Using document.querySelector('title') directly might fail in scenarios where the <title> tag is dynamically removed or recreated. Observing the head ensures the logic works even if the <title> element undergoes structural changes.